api/azure/webhook_test.go

```diff
@@ -4,116 +4,15 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-package azure_test
```

Comment:
Briefly read about the implication of using azure_test vs azure, and the difference is being that using azure_test will restrict the test from using non-exported identifiers. In the past experience with Java testing, its almost preferable to only use exported variables/functions in the tests and test from the perspective of a package user, instead of opening "back door" to allow private variables to be tested.

Reply:
> In the past experience with Java testing, its almost preferable to only use exported variables/functions in the tests and test from the perspective of a package user

That is true, but the convention in Go is to prefer having tests in the same package as the source, as seen in standard libraries and examples. In fact, Go generally requires all files in the same directory to be in the same package (it'll complain otherwise). Since Go differentiates exported/private names by capitalization, it should be pretty clear if you are accessing/testing a private method in a test. In other words, it won't happen "by accident" as in Java, but rather intentionally (authors can make the judgement call to depend on private stuff in tests if necessary).

`pacakge_test` is really a very special case in Go -- the only time Go does not complain when there are multiple packages in the same directory. It has legitimate uses, to enforce black-box testing, and more commonly to work around cyclic dependency (e.g. `util` depends on `shared`, and you're writing a test for `shared` that depends on `util`).